web/web.go

```diff
@@ -291,6 +292,10 @@ func New(logger log.Logger, o *Options) *Handler {
 		ready: 0,
 	}
 
+	if h.tsdb != nil {
```

Comment:
Why would tsdb ever be nil?

Reply:
probably when cortex imports web

Reply:
Seems unlikely as it'd be panicing a lot, plus that's not a good reason not to simplify the code.

Reply:
It happens a lot in the test code here https://github.com/prometheus/prometheus/blob/master/web/web_test.go. TSDB is not initialized here. Does it make sense to you or I should initialize TSDB in the test code?

Reply:
I'd update the test code. This logic is already fairly intricate without making more complex.

Reply:
OK. Let me update it.

Reply:
Updated. Please take a look. Thanks!